src/auth/RequireAuth.tsx

```
@@ -68,9 +71,12 @@ const RequireAuth: React.FC<RequireAuthProps> = ({ children, LoginPage }) => {
     return <LoginPage />;
   }
 
-  // In the future optimistically skip waiting if we have cached auth data
-  // TODO remove isDeprecatedAuthLoading when useGetAuthQuery is no longer used
-  if ((isLoading && !isLoggedIn) || isDeprecatedAuthLoading) {
+  if (error && ErrorPage) {
```

I think there's an error in the logic. If getMe fails with an error, the component will currently render the children, which will then probably error out.

I think if ErrorPage is not defined, this component may need to re-throw the error so that it goes up to the ErrorBoundary

---

I agree on the error handling approach.

The `ErrorPage` here was moved from the options `App` component. Would be nice to refactor within the scope of #2937